judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java

```diff
@@ -473,4 +482,60 @@ void testCompositionWithUndefinedRequiredFields() {
+        assertNotEquals(singleConA.identifier().getIdentifier() ,entityA.getSingleConA().orElseThrow().identifier().getIdentifier());
+        assertNotEquals(singleRequiredConA.identifier().getIdentifier() ,entityA.getSingleRequiredConA().identifier().getIdentifier());
+        assertEquals("TEST-A", entityA.getStringA().orElseThrow());
+
+        Collection<EntityD> ds = entityA.getSingleRequiredConA().getMultipleDonB();
+
+        EntityD testD1 = ds.stream().filter(d -> d.getStringD().orElseThrow().equals("D1")).findFirst().orElseThrow();
+        EntityD testD2 = ds.stream().filter(d -> d.getStringD().orElseThrow().equals("D2")).findFirst().orElseThrow();
+        assertNotEquals(entityD1.identifier().getIdentifier(), testD1.identifier().getIdentifier());
+        assertEquals(entityD1.getStringD().orElseThrow(), testD1.getStringD().orElseThrow());
+        assertNotEquals(entityD2.identifier().getIdentifier(), testD2.identifier().getIdentifier());
+        assertEquals(entityD2.getStringD().orElseThrow(), testD2.getStringD().orElseThrow());
+    }
+
+    @Test
+    void testDeepCopyUpdate() {
+        EntityA a2 = entityADao.create(EntityA.builder().withSingleRequiredConA(EntityC.builder().build()).build());
+        assertEquals(Optional.empty(), a2.getSingleConA());
+        assertEquals(0, a2.getCollectionConA().size());
+
+        assertEquals(5, entityCDao.countAll());
+        assertEquals(6, entityDDao.countAll());
+
+        a2.setSingleConA(EntityC.builder().withStringC("C1").build());
+        a2.setCollectionConA(List.of(EntityC.builder().withStringC("C2").withMultipleDonB(List.of(EntityD.builder().withStringD("D3").build())).build()));
+
+        entityADao.update(a2);
+
+        assertEquals(7, entityCDao.countAll());
+        assertEquals(7, entityDDao.countAll());
+        assertEquals("C1", a2.getSingleConA().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C2", a2.getCollectionConA().get(0).getStringC().orElseThrow());
+        assertEquals("D3", a2.getCollectionConA().get(0).getMultipleDonB().get(0).getStringD().orElseThrow());
+
+        EntityC c3 = entityCDao.create(EntityC.builder().withStringC("C3").build());
+        EntityC c4 = entityCDao.create(EntityC.builder().withStringC("C4").withMultipleDonB(List.of(EntityD.builder().withStringD("D3").build())).build());
+        EntityA a3 = entityADao.create(EntityA.builder().withSingleConA(c3).withSingleRequiredConA(c4).build());
+
+        assertEquals(11, entityCDao.countAll());
+        assertEquals(9, entityDDao.countAll());
+
+        a3.setSingleConA(EntityC.builder().withStringC("C3Updated").build());
+        a3.setSingleRequiredConA(EntityC.builder().withStringC("C4Updated").withMultipleDonB(List.of(EntityD.builder().withStringD("D3Updated").build())).build());
+
+        entityADao.update(a3);
+
+        assertEquals(11, entityCDao.countAll());
+        // TODO: JNG-5213 update does not create new EntityD instance
```

Comment:
I tested nested composition update and create and both work if payloads are prepared properly. TBD